src/shared/common/base_data_type.h

```diff
@@ -122,6 +124,22 @@ struct DataTypeEquivalence<TypeVec3d, enable_if_is_either_t<TypeVec3d, Vec3d, De
     using device_t = DeviceVec3d;
 };
 
+template<class TypeArray2i>
+struct DataTypeEquivalence<TypeArray2i, enable_if_is_either_t<TypeArray2i, Array2i , DeviceArray2i >> {
+    using host_t = Array2i;
+    using device_t = DeviceArray2i;
+};
+
+template<class TypeArray3i>
+struct DataTypeEquivalence<TypeArray3i, enable_if_is_either_t<TypeArray3i, Array3i , DeviceArray3i>> {
+    using host_t = Array3i;
+    using device_t = DeviceArray3i;
+};
+
```

Comment:
Variable name host_t and device_t maybe changed to host_type and device_type for clarity.

Reply:
C++ standard usually appends `_t` when referring to a type (`size_t` for instance). So I thought it would closer to the C++ jargon.

But if `_type` is clearer I can change it.

Reply:
That's ok, you can change it next PR.